### ISSUE TITLE
Add Argument option to set vagrant provider and box image

### ISSUE THREAD

**Issue by jbone81** [defenseunicorns/zarf#123]
As we look to utilize zarf to begin packaging up bigbang and the desired environment in support of multiple air gapped information systems, I think it would extremely useful to enable developers to select the vagrant provider and also the box image to be used (in particular my team is looking at RHEL or Rocky which we plan on STIG'ing to make our our SCA/ISSP(s)and AO(s).) 

I am currently running into issues with running make all in the examples and trying to get zarf ready to test in our Tanzu dev and AWS. VBox is not allowed by our IA team even for dev, so we must use vmWare Workstation or Fusion, which makes for an enjoyable time for trying to build a dev version with a connected pc or mac. 

The work everyone has done so far is amazing, thank you for your contributions on Zarf!

**Comment by RothAndrew** [comment]
Thinking out loud:

I want to start looking at the Vagrantfile situation, I'm thinking #85 goes first and is meant to be the simplification of the vagrant-based code by combining the stuff that is in the root folder with the stuff that is in the examples folder, and maybe adding some kind of framework that makes it easier to add additional dimensions of support such as other distros or hypervisors.

This issue seems more tricky to me, since none of the core Zarf team uses vmware. @jeff-mccoy and I use VirtualBox and @btlghrants uses `libvert`. We would need to identify someone who does to be the maintainer of that additional capability. @jbone81 is that something you would be interested in supporting?

**Comment by RothAndrew** [comment]
The other option is we add documentation on how to more easily extend the existing codebase to support whichever alternative technology any particular user is using, such as @btlghrants's work to do a "VM-within-a-VM so that he can do VBox based virtualization inside his `libvirt` environment.

**Comment by btlghrants** [comment]
With #84, #85, and #123 (this issue) all tied in some way to being able to run/demo Zarf in 1) disparate OSes, and 2) on different hypervisors, it seems like a good place to start would be, like you say:  "some kind of framework that makes it easier to add additional dimensions of support such as other distros or hypervisors"

The first part of that would need to be figuring out / making a statement about what we do & don't want to support for dev vs. demo.  For example, perhaps:

1. development is vbox-only -- our recommendation for others is to run vbox-in-something-else (like I do) but we don't support it (or maybe just not initially, I dunno) and

2. demo environments are available for other OS / hypervisor combos -- check the support matrix for notes (etc.).

A policy like that (at least) puts out of scope the requirement for us to support ALL the crazy dev-required, Vagrant edge cases that occur across different providers (like how they all support different synced_folder mechanisms) _while still allowing_ us to show off & test Zarf in other OSes--it's just a matter of copying Zarf in on box build (or by simple download or whatever) and needs almost _none_ of the tricky Vagrant-provider-specific stuff in this case.

A framework for _running a pre-compiled version of Zarf_ in a bunch of different OSes is much lighter to make/maintain than one to _develop Zarf_ across the same skew, after all.

**Comment by jeff-mccoy** [comment]
Or we could just really change how we think about the whole thing--I'd be super interested in a K3d (or pick your distro) + [kubevirt](https://kubevirt.io/) to do this.  We could then also get e2e doing that as discussed before.  Then the issue of vm providers goes away completely.  This would also unify testing/examples much better + enable seamless e2e local testing.

**Comment by RothAndrew** [comment]
Love it. Want to team up on something like that for the next Dash Days?

**Comment by btlghrants** [comment]
I mean... if that's in-scope, I'm game--seems hella interesting to me, tbh.

**Comment by jeff-mccoy** [comment]
This would be great for dash days--also we (Defense Unicorns) need to prove out kubevirt automation anyone for other teams, so this would be super valuable.

**Comment by RothAndrew** [comment]
Okay. I'll put this back on the shelf for now till after the next dash days. I'll look at continuing to improve the E2E testing stuff.

**Comment by jeff-mccoy** [comment]
sounds great, also some of this pain goes away too once native apply is done, because then you won't really need a VM to just test how zarf works

**Comment by jbone81** [comment]
Totally agree, I am still having issues with vmware-desktop playing nicely as a provider, but I was able to swap the vm out for a rocky8 box and that works fine, except I blew up the fs a few times once I applied the stig profile to the vm... currently going to figuring out some lessons learned about which stig items make for "fun" times when rolling out a fresh airgap deploy. 

I still think it would be a very useful thing for devs to be able to use flags to force a vagrant provider for testing, but it should be a later low hanging fruit things because there are so many other very cool things in the roadmap.

**Comment by RothAndrew** [comment]
I propose we close this, since we are working to move away from Vagrant as a recommended/supported option as we move toward just specifying Docker + KinD/K3d. Vagrant is still a valid way to run, especially if wanting to do a linux-based K3s "Applicance Mode" deployment on a dev laptop, but I'm thinking we will want to stop officially supporting that setup and just let the user use whatever they want, maybe with some suggestions/recommendations.
